src/GroupManager.php

```diff
@@ -44,13 +69,41 @@ class GroupManager {
 
   /**
    * Constructs an GroupManager object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The service providing information about bundles.
    */
-  public function __construct(ConfigFactoryInterface $config_factory) {
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeBundleInfoInterface $entity_type_bundle_info) {
     $this->configFactory = $config_factory;
+    $this->entityTypeBundleInfo = $entity_type_bundle_info;
     $this->refreshGroupMap();
   }
 
   /**
+   * Magic getter.
+   *
+   * @param string $property
+   *   The property being gotten.
+   *
+   * @return mixed
+   *   The property value.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when an invalid property is passed.
+   */
+  public function __get($property) {
```

Comment:
I'm still not sure about the getter trick. Magic getter makes sense to me when dealing with abstract entities, and we need to get things on the fly.

Wouldn't a `getGroupRelationMap` with a `static::$cache` achieve the same, but with less magic?